src/inspect_ai/model/_providers/openai.py

```diff
@@ -112,21 +112,43 @@ def __init__(
             900.0 if self.service_tier == "flex" else None
         )
 
-        # resolve api_key
+        # resolve api_key or managed identity (for Azure)
+        self.token_provider = None
         if not self.api_key:
             if self.service == "azure":
                 self.api_key = os.environ.get(
                     AZUREAI_OPENAI_API_KEY, os.environ.get(AZURE_OPENAI_API_KEY, None)
                 )
+                if not self.api_key:
```

Comment:
It seems like we should factor this and the previous similar code in azureai.py out into a common helper function.